app/containers/message/Urls.tsx

```diff
@@ -110,8 +112,18 @@ const Url = React.memo(
 		const hasContent = url.title || url.description;
 
 		let image = url.image || url.url;
+		
+		
+		
+
 		if (image) {
 			image = image.includes('http') ? image : `${baseUrl}/${image}?rc_uid=${user.id}&rc_token=${user.token}`;
+			Image.getSize(image, (width, height) => {
```

Comment:
You're doing this on every render.
Can you make it run once after image is downloaded?

Reply:
In the current implementation, the image dimensions are fetched and set once the image loads. This process triggers a state update which re-renders the component to apply the new dimensions. Here’s a step-by-step outline of what happens:

Initial Render: The component initially renders with imageLoadedState set to 'loading' and imageDimensions set to { width: 0, height: 0 }.
Image Loading: If an image URL is available, the Image.getSize method is called to fetch the image dimensions.
State Update: When the dimensions are fetched successfully, setImageDimensions and setImageLoadedState are called to update the state.
Re-render: The state update triggers a re-render of the component with the new dimensions.

Reply:
Add `console.count('PREVIEW RENDER')` on your code.
You'll see the same logs from the image.

![Screenshot 2024-07-23 at 1 34 14 PM](https://github.com/user-attachments/assets/50eb2dec-8e87-4943-9f94-26ecd65302dd)

What I asked you to do is what I did on https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/5789/commits/0324cb4490ecbb78074a9ad3f81db0335f61649b

Here's the difference.
![image](https://github.com/user-attachments/assets/88e47f44-fc61-49dc-85be-6facd0678456)